lib/galaxy/webapps/galaxy/controllers/async.py

```diff
@@ -84,12 +97,27 @@ def index(self, trans, tool_id=None, data_secret=None, **kwd):
             #
             # no data_id must be parameter submission
             #
+            GALAXY_TYPE = None
             if params.data_type:
                 GALAXY_TYPE = params.data_type
             elif params.galaxyFileFormat == 'wig':  # this is an undocumented legacy special case
                 GALAXY_TYPE = 'wig'
             else:
-                GALAXY_TYPE = params.GALAXY_TYPE or tool.outputs.values()[0].format
```

Comment:
If `GALAXY_TYPE` is provided, we don't need to do any of the looping below. Probably more straightforward to have 
```
elif params.GALAXY_TYPE:
    GALAXY_TYPE = params.GALAXY_TYPE
else: #logic below.
```